store/sqlstore/post_store.go

```diff
@@ -1684,41 +1684,36 @@ func (s *SqlPostStore) buildSearchUserFilterClause(users []string, paramPrefix s
 	clause := strings.Join(clauseSlice, ", ")
 	if byUsername {
 		if exclusion {
-			return "AND Username NOT IN (" + clause + ")", queryParams
+			return builder.Where("Username NOT IN (" + clause + ")"), queryParams
 		}
-		return "AND Username IN (" + clause + ")", queryParams
+		return builder.Where("Username IN (" + clause + ")"), queryParams
 	}
 	if exclusion {
-		return "AND Id NOT IN (" + clause + ")", queryParams
+		return builder.Where("Id NOT IN (" + clause + ")"), queryParams
 	}
-	return "AND Id IN (" + clause + ")", queryParams
+	return builder.Where("Id IN (" + clause + ")"), queryParams
 }
 
 func (s *SqlPostStore) buildSearchPostFilterClause(fromUsers []string, excludedUsers []string, queryParams map[string]interface{}, userByUsername bool, builder sq.SelectBuilder) (sq.SelectBuilder, map[string]interface{}) {
 	if len(fromUsers) == 0 && len(excludedUsers) == 0 {
 		return builder, queryParams
 	}
 
-	filterQuery := `
-		UserId IN (
-			SELECT
-				Id
-			FROM
-				Users,
-				TeamMembers
-			WHERE
-				TeamMembers.TeamId = :TeamId
-				AND Users.Id = TeamMembers.UserId
-				FROM_USER_FILTER
-				EXCLUDED_USER_FILTER)`
-
-	fromUserClause, queryParams := s.buildSearchUserFilterClause(fromUsers, "FromUser", false, queryParams, userByUsername)
-	filterQuery = strings.Replace(filterQuery, "FROM_USER_FILTER", fromUserClause, 1)
-
-	excludedUserClause, queryParams := s.buildSearchUserFilterClause(excludedUsers, "ExcludedUser", true, queryParams, userByUsername)
-	filterQuery = strings.Replace(filterQuery, "EXCLUDED_USER_FILTER", excludedUserClause, 1)
+	// Sub-query builder.
+	sb := sq.Select("Id").From("Users, TeamMembers").Where("TeamMembers.TeamId = :TeamId").Where("Users.Id = TeamMembers.UserId")
+	sb, queryParams = s.buildSearchUserFilterClause(fromUsers, "FromUser", false, queryParams, userByUsername, sb)
+	sb, queryParams = s.buildSearchUserFilterClause(excludedUsers, "ExcludedUser", true, queryParams, userByUsername, sb)
+	subQuery, _, err := sb.ToSql()
+	// This should not fail.
+	if err != nil {
+		panic(err)
```

Comment:
I am not a fan of panics, but I will leave the decision on what to do in this situation to @agnivade .

Reply:
Yes, please bubble the error upwards and handle it.